testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/openapi/service/multimodule/TestEjb.java

```diff
@@ -6,11 +6,13 @@
 package org.wildfly.test.integration.microprofile.openapi.service.multimodule;
 
 import jakarta.ejb.Stateless;
+import jakarta.enterprise.context.Dependent;
 
 /**
  * @author Joachim Grimm
  */
 @Stateless
+@Dependent
```

Comment:
Stateless EJB's must be dependent scoped.

Reply:
Just curious if this is actually needed?
In an EE environment (which I am not sure applies to MP tests?), CDI should discover EJB beans as well and basically assign its own scope if it's not explicitly added there.
See https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0#default_bean_discovery_ee

Reply:
For some reason it wasn't being discovered. Let me look into possible reasons why. I had some similar issues in the RESTEasy test suite too where I need to add a bean defining annotation. Maybe there is something more to this.

Reply:
Well, it's in an EJB jar, TBF I am not sure what rules apply in that case :thinking:

Reply:
This doesn't make it much clearer https://jakarta.ee/specifications/enterprise-beans/4.0/jakarta-enterprise-beans-spec-core-4.0#a195. I can say I still need to add the `@Dependent` annotation if the EJB is in the WAR too.

What doesn't make sense is why it works when the Jakarta REST annotations are not BDA's.

What I've noticed in debug is the `WeldDeploymentMarker.mark()` is not invoked if the annotations are not BDA's. I'm not sure in that case the `@Inject` works. I actually think the test is wrong. The resource is never invoked so you never see an NPE. However, if you invoke the endpoint you'll get an NPE.

That said, I don't know why the `@Stateless` annotations is not working as a BDA.

Reply:
Ah, that's a good link because:

> An Enterprise Beans **packaged into a CDI bean archive** and not annotated with `jakarta.enterprise.inject.Vetoed` annotation, is considered a CDI-enabled bean.

Like you say, the archive here isn't a CDI bean archive since there is nothing that would mark it as one. I think it makes sense, after all, the EJB part is an integration layer, not the core CDI.

> That said, I don't know why the `@Stateless` annotations is not working as a BDA.

It looks like it works as a BDA if it's in an archive that's a bean archive.

So adding the annotation to this test seems to make sense after all, sorry for the fuss.